packages/xrpl/src/models/utils/flags.ts

```diff
@@ -90,3 +89,30 @@ function convertFlagsToNumber(flags: GlobalFlags, flagEnum: any): number {
     return flags[flag] ? resultFlags | flagEnum[flag] : resultFlags
   }, 0)
 }
+
+/**
+ * Convert a Transaction flags property into an interface for easy interpretation.
+ *
+ * @param tx - A transaction to parse flags for.
+ * @returns An interface with all flags as booleans.
+ */
+export function parseTransactionFlags(tx: Transaction): object {
+  setTransactionFlagsToNumber(tx)
+  if (typeof tx.Flags !== 'number' || !tx.Flags || tx.Flags === 0) {
+    return {}
+  }
+
+  const flags = tx.Flags
+  const flagsMap = {}
```

Comment:
Why not use JavaScripts native `Map` class instead of an object?

Reply:
My thinking here was to keep it consistent with the parseAccountRootFlags logic in the utils, and couldn't see how much added efficiency it would add in this case vs. the extra line for exporting as an object to return. I don't feel too strongly either way, but is there a significant benefit you see for this?

Reply:
Is it possible to have duplication in flags for this case? I assumed not but that would be a hole in this logic if its the case

Reply:
Ah I see. I agree to rather not break consistency and go with using an Object then. I don't think there can be duplication using an Object, it will just overwrite the existing key entry with the new value.